src/utilities/hashed_priority_queue.zig

```diff
@@ -95,8 +95,14 @@ pub fn HashedPriorityQueue(
             try self.addUnchecked(elem);
         }
 
+        /// Returns the minimum element in the queue without removing it.
+        pub fn peek(self: *Self) !T {
+            if (self.len == 0) return Error.EmptyQueue;
```

Comment:
Why return an error in `peek()` when other methods for `HashPriorityQueue` just uses an `assert()` maybe make the error handling consistent. Probably safest to check and then return an error if a method is used incorrectly.